src/renderers/dom/shared/flattenStyleArray.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule flattenStyleArray
+ * @flow
+ */
+
+'use strict';
+
+function flattenStyleArray(nestedStylesArray: Array<Object>): Object {
```

Comment:
This is going to be a --very hot-- function, could you write it in a way that does the minimum amount of possible allocations.